components/proxy/src/test/java/com/hotels/styx/StyxServerTest.java

```diff
@@ -100,6 +107,65 @@ public void invokesPluginLifecycleMethods() {
         }
     }
 
+    @Test
+    public void adminHandlerIndicatesWhenProxyServerHasStarted() {
+        HttpClient client = simpleClient();
+
+        StyxServer styxServer = styxServer();
+        try {
+            styxServer.startAsync().awaitRunning();
+
+            assertThat(
+                    sendGet(client, "http://localhost:%s/admin/proxystatus", styxServer.adminHttpAddress().getPort()),
+                    is(""
+                            + "{\n"
+                            + "  status:UP\n"
+                            + "}"
+                            + "\n"));
+        } finally {
+            stopIfRunning(styxServer);
+        }
+    }
+
+    @Test
+    public void ifProxyFailsToStartAdminHandlerReturnsFalse() {
+        HttpClient client = simpleClient();
+
+        StyxServer styxServer = styxServerWithPlugins(failingPlugin("fail"));
+        try {
+            try {
+                styxServer.startAsync().awaitRunning();
+            } catch (Exception e) {
+                // expected
+            }
+
+            assertThat(
+                    sendGet(client, "http://localhost:%s/admin/proxystatus", styxServer.adminHttpAddress().getPort()),
+                    is(""
+                            + "{\n"
+                            + "  status:DOWN\n"
+                            + "}"
+                            + "\n"));
+        } finally {
+            stopIfRunning(styxServer);
+        }
+    }
```

Comment:
Thinking about this, we now have two states, `UP` and `DOWN`. However, if styx server fails to start, it remains indefinitely in `DOWN` state. This could be confusing for the end-users because they cannot confidently say if styx has failed to start, or if it is just slow in starting.

For this reason let's consider adding a `FAILED` state.